site/pages/api/pledge/index.ts

```diff
@@ -1,10 +1,41 @@
 import { findOrCreatePledge } from "components/pages/Pledge/lib/firebase";
 import { NextApiRequest, NextApiResponse } from "next/types";
+import { array, boolean, number, object, string } from "yup";
 
 export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse
 ) {
+  const pledgeSchema = object({
```

Comment:
Ah so this was missed in the overall comment.
Let's use the schema we've already defined for pledge form. I expect both shapes to be exactly the same so better not to duplicate this.

Only caveat I can think of is translations. We return a lingui translation id string so that probably needs to be handled on the FE.